src/rozklad_problemu.md

```diff
@@ -0,0 +1,68 @@
+# Rozklad problému
+Často se setkáte s tím, že dostanete k naprogramování úlohu, se kterou si nevíte rady a netušíte ani
+jak začít. Například:
+
+`Načti obrázek z disku, změň jeho velikost, ulož ho do jiného souboru a vykresli jej na obrazovku.`
+
+Tato úloha vypadá velmi jednoduše, když je zadaná větou (v češtině), ale obzvláště pro začínající
+programátory je obtížné převést takovouto úlohu do programovacího jazyka. Obecným pravidlem k usnadnění
+řešení složitých úloh je rozdělovat je na menší a jednodušší podúlohy tak dlouho, dokud se nedostaneme
+k podúloze, kterou již umíme vyřešit. Poté z těchto malých kousků, které máme vyřešené, zpětně poskládáme
+celý program, který vyřeší původní úlohu.
+
+Například zmíněnou úlohu můžeme rozdělit na následující podúlohy: 
+- Načti obrázek z disku
+    - Otevři soubor se vstupním obrázkem
+    - Načti hlavičku obrázku
+    - Vytvoř paměť pro pixely obrázku
+        - Naalokuj dostatek paměti dle hlavičky (šířka x výška)
+- Změň velikost obrázku
+    - Vytvoř obrázek s novým rozměrem
+    - Překopíruj původní obrázek do nového obrázku
```

Comment:
jde to pochopit jako kopirovani beze zmeny...?
do zmenseneho/zvetseneho obrazku?

Reply:
Tady jsem ani tak neměl za cíl dát funkční řešení, spíše ukázat to rozebrání na podúkoly. Logika by fungovat měla, názvosloví "Překopíruj" asi není úplně ideální, ale to už bych neřešil :)